database/migrations/2023_02_19_133321_create_carts_table.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('carts', function (Blueprint $table) {
+            $table->id();
+            $table->string('customer_email');
+            $table->string('customer_firstname');
+            $table->string('customer_lastname');
+            $table->foreignId('customer_id');
```

Comment:
refrence and nullOnDelete

Reply:
I resolve it with this code:
`            $table->foreignId('customer_id')
                ->constrained()
                ->nullOnDelete();
`